cargo-dist/src/config/mod.rs

```diff
@@ -929,25 +936,79 @@ impl std::fmt::Display for ProductionMode {
     }
 }
 
+pub(crate) fn try_load_config(
+    dist_manifest_path: Option<&Utf8PathBuf>,
+) -> DistResult<DistWorkspaceConfig> {
+    let path = dist_manifest_path.ok_or(DistError::NoDistManifest {})?;
+    load_config(path)
+}
+
+pub(crate) fn try_load_package_config(
+    dist_manifest_path: Option<&Utf8PathBuf>,
+) -> DistResult<TomlLayer> {
+    if let Some(dist_manifest_path) = dist_manifest_path {
+        let src = SourceFile::load_local(dist_manifest_path)?;
+        Ok(parse_config(src)?.dist)
+    } else {
+        Ok(Default::default())
+    }
+}
+
+pub(crate) fn load_config(dist_manifest_path: &Utf8PathBuf) -> DistResult<DistWorkspaceConfig> {
+    let src = SourceFile::load_local(dist_manifest_path.as_path())?;
+    parse_config(src)
+}
+
+pub(crate) fn parse_config(src: SourceFile) -> DistResult<DistWorkspaceConfig> {
+    let config: DistWorkspaceConfig = src.deserialize_toml()?;
+    Ok(config)
+}
+
 pub(crate) fn parse_metadata_table_or_manifest(
     manifest_path: &Utf8Path,
     dist_manifest_path: Option<&Utf8Path>,
     metadata_table: Option<&serde_json::Value>,
-) -> DistResult<DistMetadata> {
+) -> DistResult<TomlLayer> {
     if let Some(dist_manifest_path) = dist_manifest_path {
         reject_metadata_table(manifest_path, dist_manifest_path, metadata_table)?;
         // Generic dist.toml
         let src = SourceFile::load_local(dist_manifest_path)?;
-        parse_generic_config(src)
+        Ok(parse_config(src)?.dist)
     } else {
         // Pre-parsed Rust metadata table
         parse_metadata_table(manifest_path, metadata_table)
     }
 }
 
-pub(crate) fn parse_generic_config(src: SourceFile) -> DistResult<DistMetadata> {
-    let config: GenericConfig = src.deserialize_toml()?;
-    Ok(config.dist.unwrap_or_default())
+// Extremely minimal struct designed to catch the most visible difference
+// between V0 and V1 config.
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct FauxV0 {
+    #[allow(dead_code)]
+    cargo_dist_version: String,
+}
+
+#[derive(Deserialize)]
+struct FauxConfigV0 {
+    #[allow(dead_code)]
+    dist: FauxV0,
+}
+
+pub(crate) fn looks_like_v0_config(dist_manifest_path: &Utf8Path) -> bool {
+    let Ok(src) = SourceFile::load_local(dist_manifest_path) else {
+        return false;
+    };
+    src.deserialize_toml::<FauxConfigV0>().is_ok()
+}
```

Comment:
The v0 config parser is forgiving enough that it'll """successfully""" load a v1 config, with basically no contents.

So we have `FauxConfigV0` which _requires_ `dist.cargo_dist_version`, which can never be in a v1 config.

`looks_like_v0_config()` basically just makes sure `dist.cargo_dist_version` exists.

Reply:
Potential idea: flip this around so the v1 config has a `config-version` field (or something like that), and `is_v0_config()` would consist of checking that `dist.config-version` (or whatever it's called) does _not_ exist.

This would allow for cleaner config changes in the future.